client/state/data-layer/wpcom/plans/index.js

```diff
@@ -1,32 +1,62 @@
 /**
  * Internal dependencies
  */
-import wpcom from 'lib/wp';
+import { fetch } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { PLANS_REQUEST } from 'state/action-types';
 import {
-	PLANS_RECEIVE,
-	PLANS_REQUEST,
-	PLANS_REQUEST_SUCCESS,
-	PLANS_REQUEST_FAILURE
-} from 'state/action-types';
+	plansReceiveAction,
+	plansRequestFailureAction,
+	plansRequestSuccessAction,
+} from 'state/plans/actions';
 
-export const requestPlans = ( { dispatch } ) => (
-	wpcom
-		.withLocale()
-		.plans()
-		.list( { apiVersion: '1.4' } )
-		.then( plans => {
-			dispatch( { type: PLANS_REQUEST_SUCCESS } );
-			dispatch( { type: PLANS_RECEIVE, plans } );
-		} )
-		.catch( rawError => {
-			const error = rawError instanceof Error
-				? rawError.message
-				: rawError;
+/**
+ * Dispatches a request to fetch all available WordPress.com plans
+ *
+ * @param {Function} dispatch Redux dispatcher
+ * @param {Object} action Redux action
+ * @returns {Object} HTTP request action
+ */
+export const requestPlans = ( { dispatch }, action ) =>
+	dispatch( fetch( {
+		apiVersion: '1.4',
+		method: 'GET',
+		path: '/plans',
```

Comment:
> can we have some sensible defaults, so that we don’t need almost any of it?

absolutely! so far I've erred on the side of being overly explicit to make sure we don't hide "magic" in there anywhere.

already I've pretty much decided to wrap `next` in the API middleware with `local` so that we can remove that aspect of this from writing these endpoint handlers. the reason? can't seem to think of any cases where we would want to pass this along to the next piece of data-layer middleware.

also as I got into this PR I realized that it may be a fair make the `onSuccess` and `onError` optional, meaning that if they aren't there they can safely be dropped/ignored. I'm not sure what kind of default we could choose unless we also included the original action in the call to `fetch` (which I think is a wrong choice).

that leaves two/four necessary parameters: `method`, `path`, `onSuccess`, and `onError`; and I don't think this is unreasonable, but we can find out more as we explore it.

thoughts? did I address your question?